Add array slice method

creikey opened the pull request:
Closes #4715 if pythonic syntactic sugar is decided to be not needed.

creikey committed:
Add array slice method

karroffel commented on modules/gdnative/gdnative_api.json:
```diff
@@ -2933,6 +2933,17 @@
         ]
       },
+      {
+        "name": "godot_array_slice",
```
This needs to be part of the Core 1.1 extension and not placed in here, as this will cause binary compatibility issues.

creikey replied:
In my latest push I've incremented the core api version.

karroffel replied:
That's not what I meant :smile: I meant that this entry for `godot_array_slice` needs to live inside `core/next/next/api` (The Core 1.2 extension) instead of the `core/api` where it is currently located.

creikey replied:
Ahhh, ok :laughing: sorry my bad. I've moved the slice json object into the core/next/next/api section in my latest force push and gotten rid of the change in the python script.

KoBeWi commented:
If you are adding the documentation, the `deep` argument should be documented too. Also `delta` should be wrapped in `[code]` block.

Calinou commented:
Would `step` be more appropriate than `delta` as the name of the third parameter?

creikey commented:
@Calinou I've changed the name of the argument to `step` as `delta` is used in `_process`, and added an error print if zero is given as a `step`.

@KoBeWi I've explained the `deep` argument in the docs and wrapped the `delta` now `step` in a code block

akien-mga commented on modules/gdnative/gdnative_api.json:
```diff
@@ -1,22 +1,22 @@
-      "minor": 0
-    },
-    "next": {
+    "core": {
```
Why change the indentation?

akien-mga replied:
The commit log should also be amended now that it's no longer bumping core version.

creikey replied:
@akien-mga In my latest force push I've fixed getting rid of the indentation and ammended the commit log.

akien-mga commented on modules/gdnative/gdnative_api.json:
```diff
@@ -1,22 +1,22 @@
       "version": {
         "major": 1,
-        "minor": 1
+        "minor": 0
```
You're still changing the indentation, causing a diff while no actual change is made to the `core` definitions.

creikey replied:
@akien-mga I see the issue now, it was not showing up in my editor very clearly. I need to increase tab size for json... it has been fixed in the latest force push.

akien-mga commented:
Thanks!
